File: src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.CSharp.Codefix.Testing.MSTest/CodefixVerifier.cs
// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.

using Microsoft.CodeAnalysis.CodeFixes;
using Microsoft.CodeAnalysis.Diagnostics;

namespace Microsoft.CodeAnalysis.CSharp.Testing.MSTest
{
    public static class CodefixVerifier
    {
        public static CodefixVerifier<TAnalyzer, TCodefix> Create<TAnalyzer, TCodefix>()
            where TAnalyzer : DiagnosticAnalyzer, new()
            where TCodefix : CodeFixProvider, new()
        {
            return new CodefixVerifier<TAnalyzer, TCodefix>();
        }
    }
}

File: src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.VisualBasic.CodeFix.Testing.MSTest/CodeFixVerifier.vb
Imports Microsoft.CodeAnalysis.CodeFixes
Imports Microsoft.CodeAnalysis.Diagnostics

Namespace Microsoft.CodeAnalysis.VisualBasic.Testing.MSTest
    Module CodeFixVerifier
        Public Function Create(Of TAnalyzer As {DiagnosticAnalyzer, New}, TCodeFix As {CodeFixProvider, New})() As CodeFixVerifier(Of TAnalyzer, TCodeFix)
            Return New CodeFixVerifier(Of TAnalyzer, TCodeFix)
        End Function
    End Module
End Namespace
